lib/writers/git.go

```diff
@@ -95,10 +102,40 @@ func NewGitWriter(logger logr.Logger, stateStoreSpec v1alpha1.GitStateStoreSpec,
 			return nil, fmt.Errorf("password not found in secret %s/%s", stateStoreSpec.SecretRef.Namespace, stateStoreSpec.SecretRef.Name)
 		}
 
-		authMethod = &http.BasicAuth{
+		authMethod = &githttp.BasicAuth{
 			Username: string(username),
 			Password: string(password),
 		}
+	case v1alpha1.GitHubAppAuthMethod:
+		appID, ok := creds["appID"]
+		if !ok {
+			return nil, fmt.Errorf("appID not found in secret %s/%s", stateStoreSpec.SecretRef.Namespace, stateStoreSpec.SecretRef.Name)
+		}
+		installationID, ok := creds["installationID"]
+		if !ok {
+			return nil, fmt.Errorf("installationID not found in secret %s/%s", stateStoreSpec.SecretRef.Namespace, stateStoreSpec.SecretRef.Name)
+		}
+		privateKey, ok := creds["privateKey"]
+		if !ok {
+			return nil, fmt.Errorf("privateKey not found in secret %s/%s", stateStoreSpec.SecretRef.Namespace, stateStoreSpec.SecretRef.Name)
+		}
```

Comment:
nit: there's a `!ok` if statements in this file... maybe its time to use proper types? 😬 

```golang
case v1alpha1.GitHubAppAuthMethod:
  githubAppCreds, err := NewGithubAppCreds(creds)
  if err != nil {
     return nil, err
  }
```

or something like that.

Not blocking but could be nice